File: lgc-api-tp1/src/main/java/br/unitins/topicos1/lgc/Usuario/service/UsuarioService.java
package br.unitins.topicos1.lgc.Usuario.service;

import java.util.List;

import br.unitins.topicos1.lgc.Usuario.dto.UsuarioDTO;
import br.unitins.topicos1.lgc.Usuario.model.Usuario;

public interface UsuarioService {
    

    List<Usuario> findAll();
    List<Usuario> findByNome(String nome);
    Usuario findById(Long id);

    
    Usuario create(UsuarioDTO dto);
    void update(Long id, UsuarioDTO dto);
    void delete(Long id);

    
}

File: frontend/src/app/services/usuario.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Usuario } from '../models/usuario.model';

@Injectable({
  providedIn: 'root'
})
export class UsuarioService {

  private readonly apiUrl: string = 'http://localhost:8080/usuarios';

  constructor(private http: HttpClient) {}

  findAll(): Observable<Usuario[]> {
    return this.http.get<Usuario[]>(this.apiUrl);
  }

  findById(id: number): Observable<Usuario> {
    const url = `${this.apiUrl}/${id}`;
    return this.http.get<Usuario>(url);
  }

  findByNome(nome: string): Observable<Usuario[]> {
    const url = `${this.apiUrl}/find/${nome}`;
    return this.http.get<Usuario[]>(url);
  }

  create(usuario: Usuario): Observable<Usuario> {
    return this.http.post<Usuario>(this.apiUrl, usuario);
  }

  update(id: number, usuario: Usuario): Observable<Usuario> {
    const url = `${this.apiUrl}/${id}`;
    return this.http.put<Usuario>(url, usuario);
  }

  delete(id: number): Observable<void> {
    const url = `${this.apiUrl}/${id}`;
    return this.http.delete<void>(url);
  }
}